cl/recap/tasks.py

```diff
@@ -1415,3 +1420,90 @@ def mark_fq_status(fq, msg, status):
     if status == PROCESSING_STATUS.SUCCESSFUL:
         fq.date_completed = now()
     fq.save()
+
+def mark_epq_status(epq, msg, status):
+    """Mark the processing queue item as some process, and log the message.
+
+    :param pq: The ProcessingQueue object to manipulate
+    :param msg: The message to log and to save to pq's error_message field.
+    :param status: A pq status code as defined on the ProcessingQueue model.
+    """
+    if msg:
+        logger.info(msg)
+    epq.status_message = msg
+    epq.status = status
+    epq.save()
+    return epq.status, epq.status_message
+
+def decode_from_base64(content: str) -> str:
+    base64_bytes = content.encode('ascii')
+    decoded_bytes = base64.b64decode(base64_bytes)
+    return decoded_bytes.decode('ascii')
+
+@app.task(
+    bind=True, max_retries=5, interval_start=5 * 60, interval_step=10 * 60
+)
+def process_recap_email(self: Task, pk: int) -> Optional[int]:
+    start_time = now()
+    epq = EmailProcessingQueue.objects.get(pk=pk)
+    mark_epq_status(epq, "", PROCESSING_STATUS.IN_PROGRESS)
+
+    try:
+        message = epq.filepath.read()
+    except IOError as exc:
+        msg = "Internal processing error (%s: %s)." % (exc.errno, exc.strerror)
+        if (self.request.retries == self.max_retries):
+            mark_epq_status(epq, msg, PROCESSING_STATUS.FAILED)
+            return None
+        else:
+            mark_epq_status(epq, msg, PROCESSING_STATUS.QUEUED_FOR_RETRY)
+            raise self.retry(exc=exc)
+
+    parsed = json.loads(message)
+
+    # NOTE - I need to do some research into this: https://docs.aws.amazon.com/ses/latest/DeveloperGuide/receiving-email-action-lambda-event.html
+    body = decode_from_base64(parsed.get("content"))
```

Comment:
Huh. I'm curious to see how this develops. I wouldn't expect to see base64 decoding here, since we generate the API POSTs and they're supposedly in JSON, but I'm often surprised.

Reply:
I have actually taken a different direction with this thanks to my testing with Sanford Law Firm (which was quite useful here). See my comment here: https://github.com/freelawproject/awscloud/pull/1#issuecomment-876509603